src/main/java/com/rabbitmq/client/impl/recovery/AutorecoveringChannel.java

```diff
@@ -67,21 +67,26 @@ public void close() throws IOException, TimeoutException {
         try {
             delegate.close();
         } finally {
-            for (String consumerTag : consumerTags) {
-                this.connection.deleteRecordedConsumer(consumerTag);
-            }
-            this.connection.unregisterChannel(this);
+            recoveryCleanup();
         }
     }
 
     @Override
     public void close(int closeCode, String closeMessage) throws IOException, TimeoutException {
         try {
-          delegate.close(closeCode, closeMessage);
+            delegate.close(closeCode, closeMessage);
         } finally {
-          this.connection.unregisterChannel(this);
+            recoveryCleanup();
         }
     }
+    
+    private void recoveryCleanup() {
+        for (String consumerTag : consumerTags) {
+            this.connection.deleteRecordedConsumer(consumerTag);
+        }
+        // TODO what about any other recorded queues and bindings that were owned by this channel? They will now cause recovery exceptions.
```

Comment:
Queues and bindings are not "owned" by channels in any way. Consumers are channel state, however. Queues and bindings are tracked by connections (in part because queues can be owned by connections — when they are exclusive — and in part because declaring something on one channel then deleting it on another should leave no traces of that entity.

Reply:
Ya thats a bad description. What that is referencing is that any queues or bindings that were created with this channel still have the recorded entities reference this now closed channel and will throw an exception during recovery. I don't know a good way to address this other than having recorded queues/exchanges/bindings use a special "recovery" channel instead of the original creating channel. I assume the only recorded items that would require to be tied to the original channel that created them would be consumers which are cleaned up.

Reply:
removed this change.